packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2849,4 +2844,31 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.inAbstractClass = oldInAbstractClass;
       }
     }
+
+    tsParseAbstractDeclaration(
+      node: any,
+    ): N.ClassDeclaration | N.TsInterfaceDeclaration {
+      const isClass = this.match(tt._class);
+      if (!isClass && !this.isContextual("interface")) {
+        this.unexpected(null, tt._class);
+      }
+      node.abstract = true;
+      if (isClass) {
+        return this.parseClass(
+          (node: N.ClassDeclaration),
+          /* isStatement */ true,
+          /* optionalId */ false,
+        );
+      } else {
+        // for invalid abstract interface
+        this.raise(
+          node.start,
+          TSErrors.NonClassMethodPropertyHasAbstractModifer,
+        );
+        this.next();
+        return this.tsParseInterfaceDeclaration(
```

Comment:
`interface` should not have a proceeding line break:
```ts
interface
Foo {}
```
is invalid. So it does not suffice to conclude that it is an interface declaration when we see a contextual `interface` name.